app/assets/javascripts/settings/ldap-conn-test.js

```diff
@@ -1,117 +1,155 @@
-    var file = $('#dex_connector_ldap_certificate').val()
-    if(!file) {
-      ldapConnectionTestPart2( $('#dex_connector_ldap_current_cert').val() )
-      return;
+  function ldapConnectionTestPart2(cert) {
+    var host = $('#dex_connector_ldap_host').val();
+    var port = $('#dex_connector_ldap_port').val();
+    var startTLS = $('#dex_connector_ldap_start_tls_true').parent().hasClass('active');
+    // Flip logic of #anonymous_bind_toggle since expanded div means anonymous binding is 'false'
+    var anonBind = ($('#anonymous_bind_toggle').attr('aria-expanded') === 'false');
+    var dn = $('#dex_connector_ldap_bind_dn').val();
+    var pass = $("#dex_connector_ldap_bind_pw[required='required']").val();
+    var baseDN = $('#dex_connector_ldap_user_base_dn').val();
+    var filter = $('#dex_connector_ldap_user_filter').val();
+    var paramArray = [host, port, cert, baseDN, filter];
+
+    if (!anonBind) {
+      paramArray.push(dn, pass);
     }
+
+    if (noneEmpty(paramArray)) { // Check if form inputs are filled in
+      $.post('/settings/ldap_test', {
+        host: host,
+        port: port,
+        start_tls: startTLS,
+        cert: cert,
+        anon_bind: anonBind,
+        dn: dn,
+        pass: pass,
+        base_dn: baseDN,
+        filter: filter
+      }).done(function (data) {
+        if (data.result.test_pass) {
+          renderValidity('Sucessfully validated connection to LDAP server', true);
+        } else {
+          renderValidity('Validation failure. ' + data.result.message, false);
+        }
+      }).fail(function (status) {
+        var valMessage = 'Failed to connect to Velum server with following error code/message: ' + status.status + '; ' + status.statusText;
+        renderValidity(valMessage, false);
+      });
+    } else {
+      renderValidity('Missing form data', false);
+    }
+  }
+
+  function ldapConnectionTestPart1() {
+    var file = $('#dex_connector_ldap_certificate').val();
     var fileInput = $('#dex_connector_ldap_certificate')[0];
     var reader = new FileReader();
 
-    reader.onload = function(e) {
-      ldapConnectionTestPart2(reader.result);
+    if (!file) {
+      ldapConnectionTestPart2($('#dex_connector_ldap_current_cert').val());
+      return;
     }
 
-    reader.readAsText(fileInput.files[0]);	
+    reader.onload = function () {
+      ldapConnectionTestPart2(reader.result);
+    };
+
+    reader.readAsText(fileInput.files[0]);
   }
 
-  function ldapFormSubmit (event) {
-    var file = $('#dex_connector_ldap_certificate').val()
-    if(!file) {
-      var currentCert = $('#dex_connector_ldap_current_cert').val();
-      if(currentCert) {
+  function ldapFormSubmit() {
+    var file = $('#dex_connector_ldap_certificate').val();
+    var currentCert;
+
+    if (!file) {
+      currentCert = $('#dex_connector_ldap_current_cert').val();
+      if (currentCert) {
         $('#dex_connector_ldap_certificate').remove();
       }
     }
+
     return true;
```

Comment:
Indeed. We didn't catch that during the review of the original [PR](https://github.com/kubic-project/velum/pull/648/files#diff-020e56db9898f48d1af5d5936701c0f4R45). I was focused on satisfying the linter instead of fixing stuff.